### ISSUE TITLE
Sync Certain Media? Or an Option for Relative Filepaths?

### ISSUE THREAD

**Issue by jdcarls2** [vassbo/freeshow#214]
I have a few machines set up to use **sync**. Pretty much everything works as expected, but I noticed that if I use a file in an **Overlay** (in this case, the logo for our church), the sync file stores the absolute filepath `C:\\Users\\etc`. This means that as I go from computer to computer, the overlay only has the image on the last machine it was configured with.

I'd like to be able to set up the overlay on one machine and use it elsewhere. I don't know that there's a simple solution for this, but I always store my things in subdirectories of the app folder, like `~/Documents/FreeShow/Media/Logo.svg`. I wonder if there were a way to designate a folder as having a relative filepath? So if each machine had a `./Media/` folder, it could look _there_ for the file, instead of an absolute path that only applies to one machine.

**Comment by vassbo** [comment]
Good suggestion. I'll see what I can do. 😊

**Comment by DrippingDumpsterJuice** [comment]
what about something like the smart folders feature in propresenter? they have two options for media, a regular managed (by the PP software) and a smart folder, which functions on the surface similar to freeshow media. The difference is that in PP the smart folders are specified by the user, then the program only has to link to the item in the folder and not a relative or absolute path. I like the current implementation in freeshow, but adding custom folders that could be linked the same way on any machine would be useful.

**Comment by vassbo** [comment]
Okay, thanks. I'll think of a good way to implement this.

**Comment by vassbo** [comment]
In version 0.9.4 I have added "Media path ID" in the cloud settings! 🤩

Having the same ID means the exact same path will sync, changing it to something else will make all newly added slide backgrounds unique.
The media path will get a random ID when adding a keys file, so if you already have that, just rename it to something unique on the different computers.

I also added an auto find feature. So if a slide background is missing, it will search through the media folders (and their folders) added in the "Media" drawer tab, if it finds a file with the same file name, it will automatically change the background to that! 😊
